### PR TITLE
Catch possible null value error in ModelFilter

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because is a patch.

## Changelog

```markdown
### Fixed
- Catch possible null value error in `ModelFilter`
```

Got the error when using the `ModelFilter` in combination with a `ModelAutocompleteType`. 

The error might show up if you bookmark a link (e.g. /list?filter[user][value]=entityID123) and you than remove the referenced entity. The `hasValue` check passes, because you provide a valid value, but there is no entity so you receive a null error.

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
Are you able to add a functional test for this bug ?

File: src/Filter/ModelFilter.php [hunk 1]
```diff
@@ -147,6 +147,10 @@ private function buildInExpression(ProxyQueryInterface $query, string $alias, Fi
         $orX = $queryBuilder->expr()->orX();
 
         foreach ($data->getValue() as $value) {
+            if (null === $value) {
```
Comment (VincentLanglet) [inline src/Filter/ModelFilter.php]: Wouldn't even safer to user `if (!\is_object($value)) {` ?
image

**Comment by VincentLanglet** [comment]
You can rebase 4.x to fix the ci

**Comment by core23** [comment]
Done

**Comment by VincentLanglet** [comment]
Thanks, I dont have a computer until next week. It's on my todo to review/merge/release this

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Done in https://github.com/sonata-project/SonataDoctrineORMAdminBundle/releases/tag/4.16.1

**Pull request merged**